addons/web/static/src/scss/ribbon.scss

```diff
@@ -55,3 +55,13 @@
         }
     }
 }
+
+// after ribbon widget there can be field widgets or oe_title which may
+// have widgets on right end, add margin-right so ribbon not overlap on it
+.o_ribbon {
+    &:not(.o_invisible_modifier) {
+        ~ .oe_title, ~ .o_field_widget {
```

Comment:
I'm not sure the tilde operator is what you want

Reply:
Here tilde is intentionally used as oe_title is before o_ribbon so to apply css if o_ribbon and oe_title share same parent, same o_field_widget